ast/compile_test.go

```diff
@@ -5053,6 +5053,82 @@ func TestRewriteDeclaredVars(t *testing.T) {
 	}
 }
 
+func TestCompileUnusedAssignedVarsErrorLocations(t *testing.T) {
+	tests := []strictnessTestCase{
+		{
+			note: "one of the two function args is not used - issue 5602 regression test",
+			module: `package test
+			func(x, y) {
+				x = 1
+			}`,
+			expectedErrors: Errors{
+				&Error{
+					Code:     CompileErr,
+					Location: NewLocation([]byte("x = 1"), "", 3, 5),
+					Message:  "assigned var y unused",
+				},
+			},
+		},
+		{
+			note: "multiple unused argvar in scope - issue 5602 regression test",
+			module: `package test
+			func(x, y) {
+				input.baz = 1
+				input.test == "foo"
+			}`,
+			expectedErrors: Errors{
+				&Error{
+					Code:     CompileErr,
+					Location: NewLocation([]byte("input.baz = 1"), "", 3, 5),
+					Message:  "assigned var x unused",
+				},
+				&Error{
+					Code:     CompileErr,
+					Location: NewLocation([]byte("input.baz = 1"), "", 3, 5),
+					Message:  "assigned var y unused",
+				},
+			},
+		},
+		{
+			note: "some unused argvar in scope - issue 5602 regression test",
+			module: `package test
+			func(x, y) {
+				input.test == "foo"
+				x = 1
+			}`,
+			expectedErrors: Errors{
+				&Error{
+					Code:     CompileErr,
+					Location: NewLocation([]byte("input.test == \"foo\""), "", 3, 5),
+					Message:  "assigned var y unused",
+				},
+			},
+		},
+		{
+			note: "wildcard argvar that's ignored - issue 5602 regression test",
+			module: `package test
+			func(x, _) {
+				input.test == "foo"
+				x = 1
+			}`,
+			expectedErrors: Errors{},
+		},
```

Comment:
I'd like to suggest an additional test that is most likely already covered elsewhere, but it doesn't hurt to be explicit:

```golang
{
	note: "argvar not used in body but in head - issue 5602 regression test",
	module: `package test 
	func(x) := x { 
		input.test == "foo" 
	}`,
	expectedErrors: Errors{},
},
```

Reply:
added, thanks!